### DISCUSSION TITLE
Update polyline on reorder

### DISCUSSION THREAD

**Discussion by Dima-Kal** [valhalla/valhalla#3687] [Q&A] [answered]
Hi, I've been trying to achieve something, I've been through the docs and I'm not sure if I've missed something or not so i came here for help:

I have an interface for planning a route, but i also allow the user the reorder the locations after the result is provided, but when i reorder the result, i also want to update the polyline shape (because obviously the previous shape is no longer relevant), what i've tried to do is to use the turn-by-turn api to send all the locations (after they've been altered) and get a new polyline shape (because there isn't any route optimization needed, just get a turn by turn navigation for multiple stops) but the problem is that i only get a shape for each leg, and it would be a nightmare to decode each leg and combine them to one polyline and encode again. what would be great is to get a summary polyline of all the legs, is that even possible?

Am i approaching this wrong?

**Comment by dnesbitt61** [comment, marked as answer]
If you set a location type to via or through then it should keep 1 leg in the route.

**Reply by Dima-Kal** [reply]
Works, thank you very much!
